pkg/registry/core/trace/ns_registry.go

```diff
@@ -41,93 +41,93 @@ type traceNetworkServiceRegistryFindClient struct {
 
 func (t *traceNetworkServiceRegistryFindClient) Recv() (*registry.NetworkService, error) {
 	operation := typeutils.GetFuncName(t.NetworkServiceRegistry_FindClient, "Recv")
-	span := spanhelper.FromContext(t.Context(), operation)
-	defer span.Finish()
+	log, ctx := logger.NewSpanTrace(t.Context(), operation)
+	defer log.Close()
 
-	ctx := withLog(span.Context(), span.Logger())
+	ctx = logger.WithLog(ctx, log)
 	s := streamcontext.NetworkServiceRegistryFindClient(ctx, t.NetworkServiceRegistry_FindClient)
 	rv, err := s.Recv()
 	if err != nil {
 		if _, ok := err.(stackTracer); !ok {
 			err = errors.Wrapf(err, "Error returned from %s", operation)
-			span.LogErrorf("%+v", err)
+			log.Errorf("%+v", err)
 			return nil, err
 		}
-		span.LogErrorf("%v", err)
+		log.Errorf("%v", err)
 		return nil, err
 	}
-	span.LogObject("response", rv)
+	log.Info("response", rv)
 	return rv, err
 }
 
 func (t *traceNetworkServiceRegistryClient) Register(ctx context.Context, in *registry.NetworkService, opts ...grpc.CallOption) (*registry.NetworkService, error) {
 	operation := typeutils.GetFuncName(t.traced, "Register")
-	span := spanhelper.FromContext(ctx, operation)
-	defer span.Finish()
+	log, ctx := logger.NewSpanTrace(ctx, operation)
+	defer log.Close()
 
-	ctx = withLog(span.Context(), span.Logger())
+	ctx = logger.WithLog(ctx, log)
```

Comment:
Why we need it, logger should already return context with itself with logger.NewLogger(ctx, operation)